### ISSUE TITLE
jsdoc/valid-types makes "@this {object}" to fail

### ISSUE THREAD

**Issue by yvele** [gajus/eslint-plugin-jsdoc#449]
I have a function that should be called this a `this` referring any object (not `null` not `undefined`)

I have this JSDoc:

```js
/**
 * @this {object}
 */
function foo() {}
``` 

But `eslint-plugin-jsdoc` is failing with this error:
 
> Tag @this must have either a type or namepath  jsdoc/valid-types

Is this a bug or should I use something else than `@this {object}`? 🤔

**Comment by yvele** [comment]
Ok my bad it is supposed to be `@this object` 😅 

https://jsdoc.app/tags-this.html

**Comment by brettz9** [comment]
In Closure it expects the bracket, but not in regular jsdoc; if you are supporting closure, set `settings.jsdoc.mode` to "closure". We might need to change the behavior for TypeScript (or file an issue with TypeScript) though because it lists the brackets too at https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html#this .
